### ISSUE TITLE
Notification service extention for IOS Push Notifications 

### ISSUE THREAD

**Issue by danibarzo** [thunkable/thunkable-issues#759]
The notification service extention setup is required from IOS platform in order you can show an image when using Push Notifications feature 

Would be nice if thunkable can have a procedure to setup the notification service extention without using a Mac with a new version of Xcode which is a requirement now. Just as the procedure Thunkable already have to publish IOS APPs (using only a Windows PC) 

**this is needed only for IOS 

Thanks in advance!

**Comment by MrMazzone** [comment]
Is this with OneSignal?  I did notice that iOS users cannot receive the image in the notification.  Is that what you are referring to?

**Comment by danibarzo** [comment]
With OneSignal and Yes, that is exactly what i am talking about @MrMazzone.

**Comment by danibarzo** [comment]
According with OneSignal support guys not seeing the Images on iOS notifications generally happens when the Notification Service Extension for your app is not setup, or is not set up correctly. I don't understand how to setup but this is the guide that OneSignal has: https://documentation.onesignal.com/docs/ios-sdk-setup. And one requirement is A Mac with a new version of Xcode.

**Comment by MrMazzone** [comment]
@danibarzo interesting.  Yes, I would love this fixed too!

**Comment by josmas** [comment]
hi @danibarzo and @MrMazzone thanks for the report and added info. I am not sure if this is a limitation for iOS; we do all the configuration in that link for all apps, so there should not be a need to have a mac and xcode (we do that already) but it might be possible that the step to setup images correctly can only be done on a one app at a time basis, in which case we won't be able to do it. I'll have a go at it in any case and report back, although most likely not until some time late next week (or even the one after because of the holidays). Thanks!

**Comment by danibarzo** [comment]
Hi Joss, currently this setup is not covered by thunkable so please help us
with a solution, you know one push notification without an image is like
nothing..

**Comment by danibarzo** [comment]
Hi @josmas, please have you checked this issue?. Thanks in advance!

**Comment by josmas** [comment]
It is on my TODO list for this week; hopefully I can get to it. Thanks!

**Comment by danibarzo** [comment]
Great, thanks!

**Comment by josmas** [comment]
Just as a heads up; we couldn't find anything wrong with the configuration of the module, so we are going to do a full upgrade as we are a couple of versions behind. The upgrade has not been scheduled yet though. Thanks!

**Comment by danibarzo** [comment]
Hi Jos, what do you mean?. I am testing and none iPhone receives the image
attached with the push notification (having set the push notification
feature with thunkable of course), only arrives the text. Have you tested
it?

**Comment by josmas** [comment]
We can see that it fails, but our config follows the docs so we don't know why it fails. We will do a full upgrade of the library to see if the problem is in the library itself. Does that make sense?

**Comment by danibarzo** [comment]
Yes of course, i am just thinking when i followed the steps to set up the Push notification feature in thunkable, the documentation never mentioned the "Notification service extension" which is actually a requirement to use images with this feature on IOS platform. But thanks for your replay and explanation, i hope we can use the complete feature soon.

**Comment by josmas** [comment]
Yes, we do that config on our side. In the same way that you only have to select a platform but  not install the SDK for that platform. We hide as much stuff as possible to make it simpler, although with Apple not a lot of things are simple (with certs and so forth)>

**Comment by danibarzo** [comment]
ok thanks, clearer. I'll be waiting for the full upgrade of the library.

**Comment by danibarzo** [comment]
Hi dear @josmas. Do you have any update about this issue?. Thanks in advance!

**Comment by josmas** [comment]
Still hasn't been scheduled, unfortunately.

**Comment by danibarzo** [comment]
Hi dear @josmas. Do you have any update about this issue?. Thanks in advance!

**Comment by danibarzo** [comment]
Hi dear @josmas . any update with this issue? at the moment images continue not arriving to IOS devices. Thanks!

**Comment by josmas** [comment]
Hi @danibarzo, apologies for the late reply. Are you having any other issues with notifications? We have just one report saying that notifications are not working on iOS, but it seems to be isolated.

**Comment by danibarzo** [comment]
hi @josmas . Yes, actually i have checked notifications are not working on IOS but i think this is an isolated case from the images issue.

**Comment by josmas** [comment]
Hi @danibarzo yes, they are independent, but can you confirm none of your iOS notifications work?
We are having some issues determining if this is happening to everyone or just a few users.

**Comment by danibarzo** [comment]
i realized the las friday that notifications didnt arrived to my iPhone i use for test the APPs. i checked in OneSignal these two issues but i dont know if they are related: 

![image](https://user-images.githubusercontent.com/68163526/118690133-5ceac300-b7cd-11eb-88b6-3f18dfea4700.png)
![image](https://user-images.githubusercontent.com/68163526/118690461-b3580180-b7cd-11eb-8ba6-f224c4b0834c.png)

So i followed the OneSignal suggestion and generated a new .p12 certificate. Now i am waiting that Apple Store publish my new APP version, after that i will be able to test if the notifications arrive or the issue remains.

**Comment by MrMazzone** [comment]
@josmas My iOS notification are working as of this morning.  However, I still have the image issue.

**Comment by weihuali0509** [comment]
@MrMazzone is yours a recent new built/published? within a week?

**Comment by MrMazzone** [comment]
@weihuali0509 Build was maybe two to three weeks ago.  So probably a pretty recent break.

**Comment by josmas** [comment]
thanks @danibarzo and @MrMazzone that is very helpful. We are also waiting here for test apps to fully determine the cause. Thanks!

**Comment by weihuali0509** [comment]
@MrMazzone can you please publish again and see if you have the APN issue?

**Comment by danibarzo** [comment]
Thank you @josmas @weihuali0509 . Please let us know if you find what is happening.

**Comment by MrMazzone** [comment]
@weihuali0509 Sure.  This error shows in OneSignal?  Some directions to reproduce would be helpful.

**Comment by danibarzo** [comment]
hi @MrMazzone i got the errors from OneSignal (the prints i sent). Now you just please puplish your APP again and try to send a push notification, then if you dont recieve the Notification on your IOS device check in OneSignal if you have this similar issue in the user:

![image](https://user-images.githubusercontent.com/68163526/118711412-3042a580-b7e5-11eb-9731-b1ea3c112825.png)

**Comment by MrMazzone** [comment]
@josmas @weihuali0509 I am getting the same error for my build from today.  Users on old build successfully received notification. 
![image](https://user-images.githubusercontent.com/40954645/118739458-86c8d780-b817-11eb-8303-dc791ad2aa0b.png)

**Comment by danibarzo** [comment]
hi @weihuali0509 , do you have any update about this issue?. Thanks in advance!

**Comment by danibarzo** [comment]
hi @weihuali0509  , do you have any update about why the push notifications are not arriving to IOS devices?. Thanks!

**Comment by weihuali0509** [comment]
The push notification should be fixed in iOS now.

**Comment by weihuali0509** [comment]
We are still evaluating the notification service extension.

**Comment by danibarzo** [comment]
thanks @weihuali0509, please let us know when you solve the Notification Service Extension issue, you know a push notification with an image is great.

**Comment by danibarzo** [comment]
hi @weihuali0509, please how about the Notification Service Extension issue, have you check it?. Thanks in advace!

**Comment by danibarzo** [comment]
hi @weihuali0509, do you have any update about the Notification Service Extension issue?. Thanks!
